sdk/go/pulumi/mocks.go

```diff
@@ -348,6 +348,12 @@ func (m *mockMonitor) RegisterStackInvokeTransform(ctx context.Context, in *pulu
 	panic("not implemented")
 }
 
+func (m *mockMonitor) RegisterLifecycleHook(ctx context.Context, in *pulumirpc.RegisterLifecycleHookRequest,
+	opts ...grpc.CallOption,
+) (*emptypb.Empty, error) {
+	panic("not implemented")
+}
```

Comment:
It looks like it but just being paranoid based on recent breakages -- we've not added this new method to any interface that existing tests/consumers would have to implement when we release, right?